### PR TITLE
[1.10.X] Add Block.getSoundType(IBlockAccess, BlockPos, Entity)

### PR DESCRIPTION
**Pull request by Madsthunder**
All this does is add an extra method that modders can use to transform what sounds are to be used.

### CODE REVIEW

**Comment by Madsthunder** [comment]
Waiting for the ban hammer to drop...

File: patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch [hunk 1]
```diff
@@ -122,7 +122,22 @@
...
+             int l = MathHelper.func_76128_c(this.field_70161_v);
+-            IBlockState iblockstate = this.field_70170_p.func_180495_p(new BlockPos(j, k, l));
++            BlockPos pos = new BlockPos(j, k, l);
++            IBlockState iblockstate = this.field_70170_p.func_180495_p(pos);
+ 
+             if (iblockstate.func_185904_a() != Material.field_151579_a)
+             {
+-                SoundType soundtype = iblockstate.func_177230_c().func_185467_w();
++                SoundType soundtype = iblockstate.func_177230_c().getSoundType(this.field_70170_p, pos, this);
+                 this.func_184185_a(soundtype.func_185842_g(), soundtype.func_185843_a() * 0.5F, soundtype.func_185847_b() * 0.75F);
```
Comment (williewillus) [inline patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch]: just instantiate a new pos. doing this raises the patch size for no reason

File: patches/minecraft/net/minecraft/entity/passive/EntityHorse.java.patch [hunk 2]
```diff
@@ -1,22 +1,52 @@
...
 +++ ../src-work/minecraft/net/minecraft/entity/passive/EntityHorse.java
-@@ -441,6 +441,7 @@
+@@ -400,12 +400,13 @@
+                 }
+             }
+ 
+-            IBlockState iblockstate = this.field_70170_p.func_180495_p(new BlockPos(this.field_70165_t, this.field_70163_u - 0.2D - (double)this.field_70126_B, this.field_70161_v));
++            BlockPos pos = new BlockPos(this.field_70165_t, this.field_70163_u - 0.2D - (double)this.field_70126_B, this.field_70161_v);
++            IBlockState iblockstate = this.field_70170_p.func_180495_p(pos);
+             Block block = iblockstate.func_177230_c();
```
Comment (williewillus) [inline patches/minecraft/net/minecraft/entity/passive/EntityHorse.java.patch]: same here. splitting the pos out saves you a couple nanoseconds and increases the patch size

**Comment by williewillus** [comment]
You might want to warn on the javadoc that the "pos" might not actually have the block in that position, because it could be arbitrary (e.g. the RenderGlobal break effect could be called by mods at any time on any pos)

Also, @Deprecate the old vanilla one and add a note saying there's a forge one.

Otherwise I think it's good :D

**Comment by Madsthunder** [comment]
Ah, I didn't see that you posted a pull request

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 3]
```diff
@@ -1332,6 +1332,16 @@
 +    public void addInformation(ItemStack stack, EntityPlayer player, List<String> tooltip, boolean advanced)
 +    {
 +    }
++    
++    /**
++     * @param worldIn The IBlockAccess the Block exists in
++     * @param pos The position of the Block
++     * @param entityIn The Entity requesting the SoundType
```
Comment (williewillus) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: Warn that pos could potentially not have the block actually there

I think the entity @param would be better like "The Entity falling on/breaking/stepping on/digging/placing this block, or null if the context has no entity"

**Comment by williewillus** [comment]
I closed mine, this one is fine

**Comment by Madsthunder** [comment]
Primary reason I wanted this was because Metallurgic Extra's BlockOreGround (https://github.com/Madsthunder/Metallurgic-Extras/blob/1.10.X/src/main/java/continuum/metalextras/blocks/BlockOreGround.java) and my experimental Multipart (https://github.com/Madsthunder/CTMultipart/blob/1.10.X/src/main/java/continuum/api/multipart/BlockMultipart.java) would have derpy code for break sounds

**Comment by williewillus** [comment]
Looks good :D

**Comment by loordgek** [comment]
@Madsthunder , amadornes is working on a mulipart api that is planned to go into forge

**Comment by cpw** [comment]
Wow, @Madsthunder can I suggest you put the git down, go read a few webpages and books, and try again in a few weeks when you've learned to git?

**Pull request closed without merging**